src/ChunkData.cpp

```diff
@@ -46,103 +39,90 @@ cChunkData::cChunkData(cAllocationPool<cChunkData::sChunkSection> & a_Pool) :
 
 cChunkData::~cChunkData()
 {
-	#if __cplusplus < 201103L
-		// auto_ptr style interface for memory management
-		if (!m_IsOwner)
-		{
-			return;
-		}
-	#endif
-	for (size_t i = 0; i < NumSections; i++)
-	{
-		Free(m_Sections[i]);
-		m_Sections[i] = nullptr;
-	}
+	Clear();
 }
 
 
 
 
 
-#if __cplusplus < 201103L
-	// auto_ptr style interface for memory management
-	cChunkData::cChunkData(const cChunkData & a_Other) :
-		m_IsOwner(true),
-		m_Pool(a_Other.m_Pool)
+cChunkData::cChunkData(const cChunkData & a_Other):
+	m_Sections(),
+	m_Pool(a_Other.m_Pool)
+{
+	for (size_t i = 0; i < NumSections; ++i)
 	{
-		// Move contents and ownership from a_Other to this, pointer-wise:
-		for (size_t i = 0; i < NumSections; i++)
+		if (a_Other.m_Sections[i] != nullptr)
 		{
-			m_Sections[i] = a_Other.m_Sections[i];
+			m_Sections[i] = Allocate();
+			*m_Sections[i] = *a_Other.m_Sections[i];
 		}
-		a_Other.m_IsOwner = false;
 	}
+}
 
 
 
 
 
-	cChunkData & cChunkData::operator =(const cChunkData & a_Other)
+cChunkData & cChunkData::operator =(const cChunkData & a_Other)
+{
+	// If assigning to self, no-op
+	if (&a_Other == this)
+	{
+		return *this;
+	}
+	Clear();
+	for (size_t i = 0; i < NumSections; ++i)
 	{
-		// If assigning to self, no-op
-		if (&a_Other == this)
+		if (a_Other.m_Sections[i] != nullptr)
 		{
-			return *this;
+			m_Sections[i] = Allocate();
+			*m_Sections[i] = *a_Other.m_Sections[i];
 		}
+	}
+	return *this;
+}
 
-		// Free any currently held contents:
-		if (m_IsOwner)
-		{
-			for (size_t i = 0; i < NumSections; i++)
-			{
-				Free(m_Sections[i]);
-				m_Sections[i] = nullptr;
-			}
-		}
 
-		// Move contents and ownership from a_Other to this, pointer-wise:
-		m_IsOwner = true;
-		for (size_t i = 0; i < NumSections; i++)
-		{
-			m_Sections[i] = a_Other.m_Sections[i];
-		}
-		a_Other.m_IsOwner = false;
-		ASSERT(&m_Pool == &a_Other.m_Pool);
-		return *this;
-	}
 
-#else
 
-	// unique_ptr style interface for memory management
-	cChunkData::cChunkData(cChunkData && other) :
-	m_Pool(other.m_Pool)
+
+cChunkData::cChunkData(cChunkData && a_Other):
+	m_Pool(a_Other.m_Pool)
+{
+	for (size_t i = 0; i < NumSections; i++)
 	{
-		for (size_t i = 0; i < NumSections; i++)
-		{
-			m_Sections[i] = other.m_Sections[i];
-			other.m_Sections[i] = nullptr;
-		}
+		m_Sections[i] = a_Other.m_Sections[i];
+		a_Other.m_Sections[i] = nullptr;
 	}
+}
 
 
 
 
 
-	cChunkData & cChunkData::operator =(cChunkData && other)
+cChunkData & cChunkData::operator =(cChunkData && a_Other)
+{
+	if (&a_Other == this)
 	{
-		if (&other != this)
-		{
-			ASSERT(&m_Pool == &other.m_Pool);
-			for (size_t i = 0; i < NumSections; i++)
-			{
-				Free(m_Sections[i]);
-				m_Sections[i] = other.m_Sections[i];
-				other.m_Sections[i] = nullptr;
-			}
-		}
 		return *this;
 	}
-#endif
+
+	if (&m_Pool != &a_Other.m_Pool)
```

Comment:
This shouldn't happen, and arguably the whole memory pool thing is badly thought out and should be redone, but I don't want to get into that argument again because we end up in the caching debate again.